### ISSUE TITLE
Migrate benchmarks to criterion?

### ISSUE THREAD

**Issue by ysimonson** [indradb/indradb#286]
This would pull in another dependency, but it's maybe worthwhile:

* The benchmarks could be run on stable
* It would generate better stats
* Results can be graphed
* It would be easier to [compare branches](https://bheisler.github.io/criterion.rs/book/user_guide/command_line_options.html#baselines)

**Comment by lightsing** [comment]
Yes, I noticed this too. Current bench method cannot seperate "setup" and "bench" code.

I'm not sure how to migrate the macro part like `full_bench_impl` to criterion.

**Comment by ysimonson** [comment]
Yeah if the two can't be combined, that would be a show-stopper.

Maybe it's possible with a combination of `OnceCell` and [`bench_with_input`](https://bheisler.github.io/criterion.rs/book/user_guide/benchmarking_with_inputs.html), where the former holds a vec of closures that construct databases with different underlying datastores. But that's pretty nasty.
